### PR TITLE
feat: Merge applicationset into argocd

### PR DESCRIPTION
**Pull request by rishabh625**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)). 

**This is first PR to merge applicationset into argocd**

fixes #8743

### CODE REVIEW

**Comment by rishabh625** [comment]
@jannfis @wtam2018  : Please 🙏  review this, it includes moving of types and register of applicationset controller here, it can't be merged as binaries of appset is not included in argocd and this contains deployment.

File: manifests/install.yaml
Comment (rishabh625) [inline manifests/install.yaml:9245]: refering to this

**Comment by rishabh625** [comment]
@crenshaw-dev : Can you please review this PR? I'll be adding newly added bitbucket support too

File: .github/workflows/ci-build.yaml
Comment (crenshaw-dev) [inline .github/workflows/ci-build.yaml:410]: Is this necessary? I think the next line starts everything from source code. I think this would only make sense if BIN_MODE were `true`.

File: cmd/argocd-notification/commands/controller.go
Comment (crenshaw-dev) [inline cmd/argocd-notification/commands/controller.go:30]: Were these changes intentional?

File: go.mod
Comment (crenshaw-dev) [inline go.mod:101]: Should this just be merged with the block above?

File: hack/update-openapi.sh
Comment (crenshaw-dev) [inline hack/update-openapi.sh:28]: What does this change do?

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
On first pass, just a few nitpicks/questions. Great work!

**Comment by crenshaw-dev** [comment]
Can you merge master? Should fix at least one of the CI failures.

File: cmd/argocd-applicationset-controller/commands/applicationset_controller.go
Comment (alexmt) [inline cmd/argocd-applicationset-controller/commands/applicationset_controller.go:38]: Please consider using `AddKubectlFlagsToSet` or `AddKubectlFlagsToCmd` from https://github.com/argoproj/argo-cd/blob/c59b8ea51de389dc8ca23017472dc176ba651668/util/cli/cli.go#L62

File: Makefile
Comment (alexmt) [inline Makefile:424]: Is this change really required ?

File: .github/workflows/ci-build.yaml
Comment (jannfis) [reply .github/workflows/ci-build.yaml:410]: This compiles and caches the Go code sequentially before starting the e2e procedure. 

Goreman does this all in parallel, saturating the GitHub runner to a point of OOM/crash.

File: util/argo/audit_logger.go
Comment (alexmt) [inline util/argo/audit_logger.go:134]: Please remove this method, since it is not yet used anywhere. Let's introduce it when we start building application set API

**Comment by alexmt** [comment]
Few minor comments from me. Overall awesome PR! Thanks a lot, @rishabh625 ! I'm eager to merge it :D

File: .github/workflows/ci-build.yaml
Comment (rishabh625) [reply .github/workflows/ci-build.yaml:410]: as jann mentioned ci checks weren't running and required this changes.

File: util/argo/audit_logger.go
Comment (rishabh625) [reply util/argo/audit_logger.go:134]: yeah sure

File: cmd/argocd-applicationset-controller/commands/applicationset_controller.go
Comment (rishabh625) [reply cmd/argocd-applicationset-controller/commands/applicationset_controller.go:38]: yes, thanks!

File: Makefile
Comment (rishabh625) [reply Makefile:424]: yes it is required for ClusterDecisionManagement tests

File: go.mod
Comment (rishabh625) [reply go.mod:101]: yeah, should be merged for cleaner go.mod

File: cmd/argocd-notification/commands/controller.go
Comment (rishabh625) [reply cmd/argocd-notification/commands/controller.go:30]: yes, on linting found it as dead code

```
cmd/argocd-notification/commands/controller.go:28:2: `defaultMetricsPort` is unused (deadcode)
	defaultMetricsPort = 9001
	^
```

Comment (rishabh625) [reply cmd/argocd-notification/commands/controller.go:30]: will fix this above change not required

File: hack/update-openapi.sh
Comment (rishabh625) [reply hack/update-openapi.sh:28]: even with GOPATH set when GO111MODULE=off , build doesn't work and failed ,but works when GO111MODULE=on

**Comment by rishabh625** [comment]
> Can you merge master? Should fix at least one of the CI failures.

done

File: cmd/argocd/commands/app.go
Comment (crenshaw-dev) [inline cmd/argocd/commands/app.go:175]: Is this related to merging applicationset?

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
Looks like maybe some unrelated changes having to do with setting a finalizer sneaked in?

**Review by crenshaw-dev — COMMENTED** [review]
Do we need to include docs/examples stuff in this PR, or save it for a follow-up PR?

**Comment by rishabh625** [comment]
> Do we need to include docs/examples stuff in this PR, or save it for a follow-up PR?

I would like to save it for next PR

We need secrets so that e2e tests for applicationset is not skipped @alexmt please 🙏 help in add those secrets here

**Comment by crenshaw-dev** [comment]
> I would like to save it for next PR

Works for me! Just checked with Alex, he agrees. :-)

**Review by crenshaw-dev — APPROVED** [review]
LGTM!

**Comment by crenshaw-dev** [comment]
Will wait for @alexmt's final review.

**Review by alexmt — APPROVED** [review]
LGTM as well!

**Pull request merged**
